### PR TITLE
Add patch files for chainer

### PR DESCRIPTION
**Pull request by ghost**
Task for #12, #13.

* Added patch for `hierarchical_softmax.py` (fixes mismatch of types)
* Added missing(old) patch for `lstm.py` (for working with ptb)
* Modified README.md

### CODE REVIEW

**Comment by ghost** [comment]
I'll wait for your review, @LWisteria

File: README.md [hunk 1]
```diff
@@ -67,9 +67,25 @@ If you want `cupy` alias, set `export CLPY_GENERATE_CUPY_ALIAS=1` before install
...
 
 Just replace `cupy` to `clpy` in your python codes and run it (e.g. `import cupy` -> `import clpy`).
-You don't need to replace codes if you install with `CLPY_GENERATE_CUPY_ALIAS`.
+You don't need to replace codes if you install with `CLPY_GENERATE_CUPY_ALIAS=1`.
 
-### tests
+### Woking with Chainer
+
+It's confirmed that ClPy works with [Chainer v3.3.0](https://github.com/chainer/chainer/tree/v3.3.0).
+There are a few patches required on Chainer codes for working with ClPy.
```
Comment (LWisteria) [inline README.md]: We're removing these patches in next release. so please emphasize that they are required for **current** ClPy.

File: README.md [hunk 2]
```diff
@@ -67,9 +67,25 @@ If you want `cupy` alias, set `export CLPY_GENERATE_CUPY_ALIAS=1` before install
...
+Please install [Chainer v3.3.0](https://github.com/chainer/chainer/tree/v3.3.0) through the following operations.
+
+```console
```
Comment (LWisteria) [inline README.md]: Could you move these detailed instructions into `patch/chainer/README` and point it from this top README?
Long README is not good.

**Review by LWisteria — CHANGES_REQUESTED** [review]
@nsakabe-fixstars Please check my comments and fix it.

**Comment by LWisteria** [comment]
I believe we can remove these when #16 was resolved.

**Comment by ghost** [comment]
I fixed it. Could you please check it?

**Comment by LWisteria** [comment]
LGTM!

**Pull request merged**